library/include/internal/hipsolver-types.h

```diff
@@ -13,6 +13,8 @@
 
 typedef void* hipsolverHandle_t;
 
+typedef void* hipsolverSyevjInfo_t;
```

Comment:
It's too late now, but we could have prevented people from mixing up `hipsolverHandle_t` and `hipsolverSyevjInfo_t` if we had defined them as pointers to opaque types, like HIP does for things like [`hiprtcProgram`](https://github.com/ROCm-Developer-Tools/HIP/blob/rocm-5.1.3/include/hip/hiprtc.h#L89). We'd also avoid the need to do all those pointer casts from `void*` to `hipsolverSyevjInfo*`.

Reply:
I'm not sure how that would work with the cuSOLVER side, since `hipsolverSyevjInfo` is only defined for rocSOLVER. I guess I would have to point to the type defined by cuSOLVER, and use the same name on the rocSOLVER side? Could become an issue if we add support for other backends.